api/diagnostics.go

```diff
@@ -518,32 +517,11 @@ func (j *DiagnosticsJob) getHTTPAddToZip(node Node, endpoints map[string]string,
 
 		j.Status = "GET " + fullURL
 		updateSummaryReport("START "+j.Status, node, "", summaryReport)
-		timeout := time.Minute * time.Duration(j.Cfg.FlagDiagnosticsJobGetSingleURLTimeoutMinutes)
-		request, err := http.NewRequest("GET", fullURL, nil)
-		if err != nil {
-			j.Errors = append(j.Errors, err.Error())
-			logrus.Errorf("Could not create a new HTTP request: %s", err)
-			updateSummaryReport(fmt.Sprintf("could not create request for url: %s", fullURL), node, err.Error(), summaryErrorsReport)
-			j.JobProgressPercentage += percentPerURL
-			continue
-		}
-		request.Header.Add("Accept-Encoding", "gzip")
-
-		client := NewHTTPClient(timeout, j.Transport)
-		resp, err := client.Do(request)
+		resp, err := get(client, fullURL)
 		if err != nil {
 			j.Errors = append(j.Errors, err.Error())
-			logrus.Errorf("Could not fetch url %s: %s", fullURL, err)
-			updateSummaryReport(fmt.Sprintf("could not fetch url: %s", fullURL), node, err.Error(), summaryErrorsReport)
-			j.JobProgressPercentage += percentPerURL
-			continue
-		}
-
-		if resp.StatusCode != http.StatusOK {
-			errMsg := fmt.Sprintf("unable to fetch %s. Return code %d", fullURL, resp.StatusCode)
-			logrus.Error(errMsg)
-			j.Errors = append(j.Errors, errMsg)
-			updateSummaryReport("wrong status code returned", node, errMsg, summaryErrorsReport)
+			logrus.Errorf("Could not read force-tls flag: %s", err)
+			updateSummaryReport(err.Error(), node, err.Error(), summaryErrorsReport)
 			j.JobProgressPercentage += percentPerURL
 			continue
```

Comment:
We don't close the body in this case; isn't this going to lead to hanging connections?

Reply:
If you close the response body here 
https://github.com/dcos/dcos-diagnostics/pull/60/files#diff-0ee9de7b9d9927cce9fb53251662d72aR564 (so if the request succeeded but the response is unexpected) then the `get()` function maintains a contract that says if it returns an error, the response body does not need to be closed. 

That's a convenient pattern to follow.

Reply:
Fixed